smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -0,0 +1,152 @@
+package com.eis.smslibrary;
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.Peer;
+
+/**
+ * Implementation of {@link Peer} for SMS communication channel.
+ *
+ * @author Luca Crema, Marco Mariotto.
+ */
+public class SMSPeer implements Peer<String>, Comparable<SMSPeer> {
+
+    public static final int MAX_TELEPHONE_NUMBER_LENGTH = 20;
+    public static final int MIN_TELEPHONE_NUMBER_LENGTH = 3;
+
+    private String telephoneNumber;
+
+    /**
+     * @param telephoneNumber Address for the peer.
+     * @throws InvalidTelephoneNumberException If telephoneNumber check is not {@link TelephoneNumberState#TELEPHONE_NUMBER_VALID}.
+     */
+    public SMSPeer(String telephoneNumber) throws InvalidTelephoneNumberException{
+        TelephoneNumberState telephoneNumberState = SMSPeer.checkPhoneNumber(telephoneNumber);
+        if(telephoneNumberState != TelephoneNumberState.TELEPHONE_NUMBER_VALID)
+            throw new InvalidTelephoneNumberException(telephoneNumberState);
+
+        this.telephoneNumber = telephoneNumber;
+    }
+
+    /**
+     * Checks if the given phone number is valid.
+     *
+     * @param telephoneNumber the phone number to check.
+     * @return The {@link TelephoneNumberState} of the telephone number after the tests.
+     */
+    public static TelephoneNumberState checkPhoneNumber(String telephoneNumber) {
+        //Check if the number is shorter than the MAX.
+        if (telephoneNumber.length() > MAX_TELEPHONE_NUMBER_LENGTH) {
+            return TelephoneNumberState.TELEPHONE_NUMBER_TOO_LONG;
+        }
+        //Check if the number is longer than the MIN.
+        if (telephoneNumber.length() < MIN_TELEPHONE_NUMBER_LENGTH) {
+            return TelephoneNumberState.TELEPHONE_NUMBER_TOO_SHORT;
+        }
+        //Check if it's actually a number and doesn't contain anything else
+        //First we have to remove the "+"
```

Comment:
Substring(1) return the string without the first char
With length -1 remove also the last char

Reply:
You can also just `substring(1)` to get from the first char to the end

Reply:
Second to last

Reply:
Yeah, you're right, sorry

Reply:
What about checking with regex if string starts with "+"? So that I can remove the substring

Reply:
In my SMSPeer I used “\\+?\\d{4,15}”
To check all in one shot if it don’t matches then I throw invalidPeerException

Reply:
With dubble\

Reply:
You don't check if the number is only numbers with that do you?

Reply:
It means
0 or 1 +
4 to 15 only number string

Reply:
Using String.matches it do it’s work

Reply:
Yup, the \+? Is searching from 0 or 1 of the + character, while \d{4,15} is searching for 4 to 15 digits in a row

Reply:
I switched to a "full" regex now, as you have suggested

Reply:
Awesome! You just need to update the comment before this line and you're done here